Dockerfile-proxy

```diff
@@ -1,17 +1,23 @@
-FROM gcr.io/linkerd-io/base:2017-10-30.01 as fetch
+ARG RUNTIME_IMAGE=gcr.io/linkerd-io/base:2017-10-30.01
 
-# XXX We must do HTTPS verification here, but we don't yet have certs in this
-# container.
-RUN curl -kvsLO https://build.l5d.io/linkerd2-proxy/latest.txt
-RUN (latest=$(awk '{print $2}' latest.txt) ; \
+# Fetches the latest proxy binary via build.l5d.io/linkerd2-proxy/latest.txt
+FROM gcr.io/linkerd-io/base:2017-10-30.01 as fetch
```

Comment:
Do you want to use the `RUNTIME_IMAGE` variable here, so that both build stages are guaranteed to have the same base?

```
FROM $RUNTIME_IMAGE as fetch
```

Reply:
The thought was that you might want to change the runtime image to something like `scratch` (or something internal to your org) -- in the build image, we require that certain tools are available (like apt, curl, etc), so i don't think it should be the same as the runtime image which doesn't have those requirements.

Reply:
Ok, that make sense. I'm curious why you pulled the runtime image out into a separate variable then, if we're only planning on using it in one place. Is it just to give it more visibility at the top of the file?

Reply:
It's sometimes convenient to be able to change on the CLI like `docker build . --build-arg RUNTIME_IMAGE=scratch`